server/data/hmppsRestClient.ts

```diff
@@ -168,4 +175,32 @@ export default class HmppsRestClient {
         throw sanitisedError
       })
   }
+
+  async postMultiPart(
+    { path = null, headers = {}, responseType = '', fileToUpload }: PostMultiPartRequest,
+    signedWithMethod?: SignedWithMethod
+  ): Promise<unknown> {
+    const signedWith = signedWithMethod?.token || (await this.tokenStore.getSystemToken(signedWithMethod?.username))
+    logger.info(`PostMultiPartRequest using admin client credentials: calling ${this.name}: ${path}`)
+    return superagent
+      .post(`${this.apiConfig.url}${path}`)
+      .attach('file', fileToUpload.path, { filename: fileToUpload.originalname, contentType: fileToUpload.mimetype })
+      .agent(this.agent)
+      .retry(2, (err, res) => {
+        if (err) logger.info(`Retry handler found API error with ${err.code} ${err.message}`)
+        return undefined // retry handler only for logging retries, not to influence retry logic
+      })
+      .auth(signedWith, { type: 'bearer' })
+      .set(headers)
+      .responseType(responseType)
+      .timeout(this.apiConfig.timeout)
+      .then(response => {
+        return response.body
+      })
+      .catch(error => {
+        const sanitisedError = sanitiseError(error)
```

Comment:
Unit test for the error handling?

Reply:
I'll take a look.

Reply:
I've added two new tests here. One for when no file is supplied and one for not found on the endpoint.